src/index.ts

```diff
@@ -1,164 +1,616 @@
+export { Session } from "./session.js"
 
-import { impl as auth } from "./auth/implementation.js"
-import { isPermissionedAppState } from "./auth/state.js"
-import { setImplementations } from "./setup.js"
-import { BASE_IMPLEMENTATION } from "./auth/implementation/base.js"
-import { USE_WNFS_IMPLEMENTATION } from "./auth/implementation/use-wnfs.js"
-import { LOBBY_IMPLEMENTATION } from "./auth/implementation/lobby.js"
-import { AppState } from "./auth/state/app.js"
-import { PermissionedAppState } from "./auth/state/permissionedApp.js"
-import * as appState from "./auth/state/app.js"
-import * as permissionedAppState from "./auth/state/permissionedApp.js"
+
+
+// TYPES & CONSTANTS
+
+
+export type AuthenticationStrategy = {
+  implementation: Auth.Implementation<Components>
+
+  accountConsumer: (username: string) => Promise<AccountLinkingConsumer>
+  accountProducer: (username: string) => Promise<AccountLinkingProducer>
+  isUsernameAvailable: (username: string) => Promise<boolean>
+  isUsernameValid: (username: string) => Promise<boolean>
+  register: (options: { username: string; email?: string }) => Promise<{ success: boolean }>
+  session: () => Promise<Maybe<Session>>
+}
+
+
+export type Program = ShortHands & {
+  auth: AuthenticationStrategy
+  components: Components
+  confidences: {
+    collect: () => Promise<Maybe<string>> // returns username
+    request: () => Promise<void>
+    session: (username: string) => Promise<Maybe<Session>>
+  }
+  session: Maybe<Session>
+}
+
+
+export enum ProgramError {
+  InsecureContext = "INSECURE_CONTEXT",
+  UnsupportedBrowser = "UNSUPPORTED_BROWSER"
+}
+
+
+export type ShortHands = {
+  loadFileSystem: (username: string) => Promise<FileSystem>
+  loadRootFileSystem: (username: string) => Promise<FileSystem>
+}
+
+
+
+// ENTRY POINTS
 
 
 /**
- * Check if we're authenticated and initiate the user's file system if 
- * authenticated (can be disabled).
+ * Build a webnative program.
  *
- * See `loadFileSystem` if you want to load the user's file system yourself.
- * NOTE: Only works on the main/ui thread, as it uses `window.location`.
+ * This will give you a `Program` object which has the following properties:
+ * - `session`, a `Session` object if a session was created before.
+ * - `auth`, a means to control the various auth strategies you configured. Use this to create sessions. Read more about auth components in the toplevel `auth` object documention.
+ * - `confidences`, a means to control confidences. Use this to collect & request confidences, and to create a session based on them. Read more about confidences in the toplevel `confidences` object documentation.
+ * - `components`, your full set of `Components`.
  *
+ * This object also has a few other functions, eg. to load a filesystem.
+ * These are called "shorthands" because they're the same functions available
+ * through other places in webnative, but you don't have to pass in the components.
```

Comment:
Ah reading below, I may not be understanding the issue here. Is it possible to directly import these helpers from a module? Or do they need to be configured, and we are really trying to avoid developers interacting with the lower-level components directly?

Reply:
> Or do they need to be configured, and we are really trying to avoid developers interacting with the lower-level components directly?

Exactly. You can import them from a module, but you will need to pass in the components yourself if you do.